### DISCUSSION TITLE
Missing endpoint resolver when deploying to Azure

### DISCUSSION THREAD

**Discussion by vicperdana** [microsoft/aspire#3965] [Q&A] [answered]
On Preview 5.

There's something obvious I may have missed, and keen to get suggestions.  When running locally, I can see the following three env vars created, including `services__chatservice__http__0`.
<img width="900" alt="image" src="https://github.com/dotnet/aspire/assets/7114832/bb64fbe8-3156-4951-8339-607070b1c0c3">

This however doesn't exist after `azd up` / deployed to Azure. 
<img width="429" alt="image" src="https://github.com/dotnet/aspire/assets/7114832/d3fb0900-7917-4543-878b-65de26e8b89f">

My AppHost `program.cs` relevant snippets
```csharp
var chatService = builder.AddProject<Projects.AspireShop_ChatService>("chatservice")
    .WithEnvironment("AzureOpenAI__ChatDeploymentName", chatDeploymentName)
    .WithEnvironment("AzureOpenAI__Endpoint", chatEndpoint)
    .WithEnvironment("AzureOpenAI__ApiKey", chatApiKey)
    .WithReference(catalogService);

builder.AddProject<Projects.AspireShop_Frontend>("frontend")
    .WithReference(basketService)
    .WithReference(catalogService)
    .WithReference(chatService)
    .WithExternalHttpEndpoints();
```

Appreciate any inputs!

**Comment by davidfowl** [comment, marked as answer]
First, update to preview 6, newer is better. Did you run `azd infra synth` at some point?

**Reply by vicperdana** [reply]
Re-running `azd infra synth` updated it which resolved the issue!  I'll upgrade right away! Thanks as always @davidfowl 🙏

**Reply by davidfowl** [reply]
https://github.com/Azure/azure-dev/issues/3782
